### PR TITLE
Adds ip-traceable listener

### PR DESCRIPTION
**Pull request by alterphp**
Adds support for IpTraceable Gedmo extension

### CODE REVIEW

**Comment by Axinet** [comment]
+1

**Comment by alterphp** [comment]
UP

**Comment by lemoinem** [comment]
:+1: Please merge this ASAP...

**Comment by ConneXNL** [comment]
+1

**Comment by guilro** [comment]
+1

File: Resources/config/listeners.xml [hunk 1]
```diff
@@ -24,6 +25,11 @@
                 <argument type="service" id="annotation_reader" />
             </call>
         </service>
+        <service id="stof_doctrine_extensions.listener.ip_traceable" class="%stof_doctrine_extensions.listener.ip_traceable.class%" public="false">
```
Comment (stof) [inline Resources/config/listeners.xml]: please do not add a parameter for the service class. This is discouraged in Symfony now

File: Resources/doc/index.rst [hunk 2]
```diff
@@ -164,10 +165,10 @@ ready, use the form component as usual. Then, after you verify the form is valid
     ;
 
     if ($this->getRequest()->getMethod() === 'POST') {
-        $form->bind($this->getRequest());
+        $form->bindRequest($this->getRequest());
 
         if ($form->isValid()) {
-            $em = $this->getDoctrine()->getManager();
+            $em = $this->getDoctrine()->getEntityManager();
```
Comment (stof) [inline Resources/doc/index.rst]: please revert these changes as they are wrong. You are switching to the old Symfony API

File: EventListener/IpTraceListener.php [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+ * IpTraceableListener
+ *
+ * @author Pierre-Charles Bertineau <pc.bertineau@alterphp.com>
+ */
+class IpTraceListener implements EventSubscriberInterface
+{
+    /**
+     * @var IpTraceableListener
+     */
+    protected $ipTraceableListener;
```
Comment (stof) [inline EventListener/IpTraceListener.php]: please make this private

File: EventListener/IpTraceListener.php [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    }
+
+    /**
+     * Set client IP from event Request
+     *
+     * @param GetResponseEvent $event
+     */
+    public function onKernelRequest(GetResponseEvent $event)
+    {
+        if (HttpKernelInterface::MASTER_REQUEST === $event->getRequestType()) {
```
Comment (stof) [inline EventListener/IpTraceListener.php]: please use an early return, and the ``isMasterRequest`` method

File: DependencyInjection/Configuration.php [hunk 5]
```diff
@@ -85,6 +86,10 @@ private function getClassNode()
                     ->cannotBeEmpty()
                     ->defaultValue('Gedmo\\Blameable\\BlameableListener')
                 ->end()
+                ->scalarNode('ip_traceable')
+                    ->cannotBeEmpty()
+                    ->defaultValue('Gedmo\\IpTraceable\\IpTraceableListener')
```
Comment (stof) [inline DependencyInjection/Configuration.php]: this setting should be removed as I don't want to introduce new class parameters anymore (it was a bad idea to introduce them)

**Comment by stof** [comment]
@alterphp please rebase your branch to fix conflict, and fix the few things I asked

File: EventListener/IpTraceListener.php (hunk 4 above)
Comment (stof) [inline EventListener/IpTraceListener.php]: I remove the part about ``isMasterRequest`` as this does not exist in 2.3

**Comment by alterphp** [comment]
@stof Done

**Comment by jewome62** [comment]
+1

**Comment by conradfr** [comment]
+1

**Comment by ConneXNL** [comment]
+1

**Comment by Invis1ble** [comment]
+ 1

**Comment by QuozPowa** [comment]
So how could we activate the ip_traceable extension with the StofDoctrineExtensionsBundle ?

**Comment by Invis1ble** [comment]
@QuozPowa 
```yml
stof_doctrine_extensions:
    orm:
        default:
            ip_traceable: true
```

**Comment by QuozPowa** [comment]
I tried but I have this error : 
Unrecognized option "ip_traceable" under "stof_doctrine_extensions.orm.default"

I use the dev-master version of stof/doctrine-extensions-bundle in composer.

**Comment by alterphp** [comment]
@QuozPowa As long as @stof does not merge this PR, you can't use it from this repository... Or you have to use my fork (I created for this PR) : git@github.com:alterphp/StofDoctrineExtensionsBundle.git on branch iptraceable-listener

If you personnaly know Stof, you can tell him to come back over here ;-)

**Comment by JulienDemangeon** [comment]
+1

**Comment by Pierstoval** [comment]
:+1: too, I think this can be merged

**Comment by ctaverne** [comment]
+1 :)

**Comment by tkuska** [comment]
+100 :)

**Comment by alterphp** [comment]
@stof, please accept that. Quicker it will be in the trunk, easier it will be to maintain...

**Comment by Neirda24** [comment]
:+1:

**Comment by ctaverne** [comment]
+1

**Comment by alterphp** [comment]
@stof I made some updates in order to enhance Symfony compliance as you told me (even if it breaks bundle uniformity). Please tell me...

**Comment by Pierstoval** [comment]
I'm still :+1: on this feature, now the implementation seems perfect I think it *really* can be merged :+1:

**Comment by speedy001** [comment]
+1

**Comment by AlanEdwardScott** [comment]
+1

**Comment by alfonsomga** [comment]
Why this has not been merged yet? any reason?

**Comment by lemoinem** [comment]
@alfonsomga Because this bundle doesn't seem to be maintained anymore...
You should take a look to https://github.com/KnpLabs/DoctrineBehaviors which offers replacements to several features of Gedmo, with a better architecture most of the time, and actual maintenance.

If you have some time, there are several projects put forward to improve the new library.

**Comment by troymccabe** [comment]
Hey @stof,

Any chance you could take a peek & hopefully merge this PR? Been a while :/

Thanks!

**Comment by ybert** [comment]
Hey @stof ,
Can you merge this PR? It could be very helpfull for us.
If you think that you will not maintain this bundle, can you give us an alternative?

Thank you for your time.

**Comment by scaytrase** [comment]
KnpLabs looks more maintained, but there are not ip_traceable too

**Comment by mikysan** [comment]
+1

**Comment by ceesvanegmond** [comment]
👍

**Comment by Guibzs** [comment]
+1

**Comment by Nek-** [comment]
@alterphp this need a rebase

@stof are you ok with this now ?

**Comment by alterphp** [comment]
Nearly 2 years since my last commit... I hope you don't mind if I wait for @stof confirmation before doing anything...

**Comment by emnsen** [comment]
why it can not be merged ?

**Comment by Nek-** [comment]
It's pretty obvious that stof has no time to maintain this repository more than just "compatiblity with new Sf versions and security fixes".

So if somebody is not happy with that, you can probably enter in discussion in order to fork the repository and maintain it. For example under friendofsymfony organization. Feel free to send some mails to take the temperature of people on the topic.

In any case more talk here is just spam for maintainer(s).

**Comment by tourze** [comment]
Any update?

**Comment by 11k** [comment]
I put [this Gist](https://gist.github.com/11k/636d54e7e28657e0856836c07d68c9f7) together to help anyone looking to integrate IpTraceable in a Symfony 5.4 app.

**Comment by stof** [comment]
Replaced by https://github.com/stof/StofDoctrineExtensionsBundle/pull/453 as the config of the bundle changed too much in the meantime.
